src/pages/_app.tsx

```diff
@@ -12,17 +12,38 @@ const PreviewProductSwitcher = dynamic(
   { ssr: false }
 )
 
-export default function App({ Component, pageProps }) {
+export default function App({ Component, pageProps, layoutData }) {
   useAnchorLinkAnalytics()
 
   const Layout = Component.layout ?? BaseLayout
 
   return (
     <ErrorBoundary>
-      <Layout>
+      <Layout {...(layoutData && { data: layoutData })}>
         <Component {...pageProps} />
       </Layout>
       {isPreview ? <PreviewProductSwitcher /> : null}
     </ErrorBoundary>
   )
 }
+
+App.getInitialProps = async ({ Component, ctx }) => {
+  const layoutQuery = Component.layout
+    ? Component.layout?.rivetParams ?? null
+    : null
+
+  const { default: rivetQuery, proxiedRivetClient } = await import('lib/cms')
```

Comment:
Is the dynamic import here to ensure the cms and all of the product data gets chunked out and not included in the client bundle?

Reply:
Yup! It generates a client-side chunk, but since `getInitialProps` is never run in the client, it doesn't get requested.